File: AndroidStudio/NewAppjava/app/src/main/java/com/aelayyuu/newapp_java/viewmodel/NewsViewModel.java
package com.aelayyuu.newapp_java.viewmodel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.aelayyuu.newapp_java.api.ApiService;
import com.aelayyuu.newapp_java.model.News;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class NewsViewModel extends ViewModel {
    MutableLiveData<News> result = new MutableLiveData<>();
    MutableLiveData<Boolean> loadError = new MutableLiveData<>();
    MutableLiveData<Boolean> loading = new MutableLiveData<>();

    public MutableLiveData<News> getResult() {
        return result;
    }

    public MutableLiveData<Boolean> getLoading() {
        return loading;
    }

    public MutableLiveData<Boolean> getLoadError() {
        return loadError;
    }

    public void loadResults() {
        loading.setValue(true);

        String apiKey = "5bb4015b85d448989fce66f1bd60de66";

        Call<News> newsCall = ApiService.getNews(
                "us", apiKey
        );
        newsCall.enqueue(new Callback<News>() {
            @Override
            public void onResponse(Call<News> call, Response<News> response) {
                if (response.isSuccessful() && response.body() != null) {
                    loading.setValue(false);
                    result.setValue(response.body());
                }

            }

            @Override
            public void onFailure(Call<News> call, Throwable t) {
                loading.setValue(false);
                loadError.setValue(true);

            }
        });
    }
}

File: AndroidStudio/API/app/src/main/java/com/wya/api/viewmodel/NewsViewModel.kt
package com.wya.api.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.wya.api.api.ApiCLient
import com.wya.api.model.News
import retrofit2.Call
import retrofit2.Response

class NewsViewModel : ViewModel() {
    private var result: MutableLiveData<News> = MutableLiveData()//assign variable
    private var errorMessage: MutableLiveData<String> = MutableLiveData()
    private var errorStatus: MutableLiveData<Boolean> = MutableLiveData()
    private var loading: MutableLiveData<Boolean> = MutableLiveData()

    fun getResult(): LiveData<News> = result//return result

    fun getErrorMessage(): LiveData<String> = errorMessage
    fun getErrorStatus(): LiveData<Boolean> = errorStatus

    fun getLoading(): LiveData<Boolean> = loading

    fun loadResult() {//set variable
        var apiCLient = ApiCLient()
        val apiCall = apiCLient.getTopHeadLines("us")
        apiCall.enqueue(object : retrofit2.Callback<News> {    //implement members in follow


            override fun onFailure(call: Call<News>, t: Throwable) {
                loading.value = false
                errorMessage.value = t.toString()
            }

            override fun onResponse(call: Call<News>, response: Response<News>) {
                loading.value = false
                errorStatus.value = false
                result.value = response.body()
            }


        })
    }
}
